projects/kit/components/rating/rating.component.ts

```diff
@@ -0,0 +1,105 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    forwardRef,
+    Inject,
+    Input,
+    Optional,
+    Self,
+    ViewChild,
+} from '@angular/core';
+import {NgControl} from '@angular/forms';
+import {
+    AbstractTuiControl,
+    isNativeFocused,
+    TUI_FOCUSABLE_ITEM_ACCESSOR,
+    tuiDefaultProp,
+} from '@taiga-ui/cdk';
+
+import {TUI_RATING_OPTIONS, TuiRatingOptions} from './rating-options';
+
+@Component({
+    selector: 'tui-rating',
+    templateUrl: './rating.template.html',
+    styleUrls: ['./rating.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {
+            provide: TUI_FOCUSABLE_ITEM_ACCESSOR,
+            useExisting: forwardRef(() => TuiRatingComponent),
+        },
+    ],
+})
+export class TuiRatingComponent extends AbstractTuiControl<number> {
+    @ViewChild('range')
+    private readonly rangeElement?: ElementRef<HTMLInputElement>;
+
+    @Input()
+    @tuiDefaultProp()
+    min = this.options.min;
+
+    @Input()
+    @tuiDefaultProp()
+    max = this.options.max;
+
+    @Input()
+    @tuiDefaultProp()
+    iconNormal = this.options.iconNormal;
+
+    @Input()
+    @tuiDefaultProp()
+    iconFilled = this.options.iconFilled;
+
+    constructor(
+        @Optional()
```

Comment:
It's not optional, it doesn't work without control.

Reply:
fixed

Reply:
@waterplea hello, unfortunately without `Optional` I have failed build

![image](https://user-images.githubusercontent.com/12021443/150111261-d4f3ab37-5b68-46cc-8498-a3ae8e5835de.png)

Reply:
Ok. Our controls historically don't break without form control. Let's keep it.